src/logger.ts

```diff
@@ -1,9 +1,26 @@
 import pino from 'pino';
 import type { Logger } from 'pino';
 
-// To prevent replays with JWT, we can redact the signature. This invalidates the JWT.
-export const redactToPreventAuthReplay = (secret: string): string =>
-	secret.replace(/^(Bearer [A-Za-z0-9]*\.[A-Za-z0-9]*\.).*/, '$1[redacted]');
+/**
+ * To prevent replays with JWT, we redact the signature. This invalidates the JWT.
+ * This is a censor function suitable for pino.redactOptions.censor.
+ * @see pino.redactOptions.censor
+ */
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-function-return-type --
+ * More closely match the (param) signature `pino` expects. Not doing so got us in trouble.
+ * Infer return type rather than define it as `any` otherwise unit tests also fail rules.
+ */
+export const redactToPreventAuthReplay = (value: any, path: string[]) => {
+	if (value === undefined) return;
+	if (typeof value === 'string')
+		return value.replace(
+			/^(Bearer [A-Za-z0-9]*\.[A-Za-z0-9]*\.).*/,
+			'$1[redacted]',
+		);
+	throw new Error(
```

Comment:
Do we truly want to error in this case?  This means we could potentially return a 500 error because of logger logic.

I'd suggest we move to gracefully returning the string as `[Unexpected value from pino]`.   We may want to move to using Sentry at some point at which point this could translate to a sentry warning since it shouldn't happen.

If we do keep this as an error then we're going to need to be one step smarter since if value is an object it will return `[object Object]` when stringified (and we have lint rules disallowing this, which makes sense). We could either check if it's an object first (and at that point not return the value at all), and only include the value if it's a primitive.

Reply:
I updated the function to guarantee that the context in which an error is thrown is the one we truly care about: authorization. This was already true before the change, but it makes it clearer.

There is a paradox where folks who want a reliable system avoid throwing errors and end up with a brittle system whereas folks who pepper the codebase with `assert` and `throw` end up with a reliable system.